api/controllers/ApplicationController.js

```diff
@@ -216,36 +216,40 @@ var applicationController = {
      */
     submitApplication: function(req, res) {
         var id = req.query.merchantReturnData;
-        console.log(id + " - attempting to submit application");
+        sails.log.info(id + " - attempting to submit application");
         Application.findOne({
             where: {
                 application_id: id
             }
         }).then(function(application) {
             if (application !== null) {
```

Comment:
there are ifs inside ifs here - as well s inside a callback which makes it hard to read. What is returned when `application` is null? 

I use "guard" clauses (you may have seen in other languages or TS). Something like this
```.js
if (application === null) {
  return "whatever"
}
// it is safe to carry on without an else, because it's already done a "return". It wont execute any further
if (application.submitted) {
//...
} else {
//...
}

if (application.serviceType == 1) {
//..
} else if (....) {
//..
}

```

Reply:
Ha, I've actually used these a lot but didn't know they were called guard clauses. You learn something new everyday :)